### PR TITLE
[AWS] Get rid of credential files if `remote_identity: SERVICE_ACCOUNT` specified

### PR DESCRIPTION
**Pull request by Michaelvll**
Tested (run the relevant ones):

- [ ] Code formatting: `bash format.sh`
- [ ] Any manual or new tests for this PR (please specify below)
- [ ] All smoke tests: `pytest tests/test_smoke.py` 
- [ ] Relevant individual smoke tests: `pytest tests/test_smoke.py::test_fill_in_the_name` 
- [ ] Backward compatibility tests: `conda deactivate; bash -i tests/backward_compatibility_tests.sh`

### CODE REVIEW

File: sky/templates/aws-ray.yml.j2 [hunk 1]
```diff
@@ -185,7 +186,12 @@ setup_commands:
     sudo grep -e '^DefaultTasksMax' /etc/systemd/system.conf || (sudo bash -c 'echo "DefaultTasksMax=infinity" >> /etc/systemd/system.conf'); sudo systemctl set-property user-$(id -u $(whoami)).slice TasksMax=infinity; sudo systemctl daemon-reload;
     {%- endif %}
     mkdir -p ~/.ssh; (grep -Pzo -q "Host \*\n  StrictHostKeyChecking no" ~/.ssh/config) || printf "Host *\n  StrictHostKeyChecking no\n" >> ~/.ssh/config;
-    [ -f /etc/fuse.conf ] && sudo sed -i 's/#user_allow_other/user_allow_other/g' /etc/fuse.conf || (sudo sh -c 'echo "user_allow_other" > /etc/fuse.conf'); # This is needed for `-o allow_other` option for `goofys`;
+    [ -f /etc/fuse.conf ] && sudo sed -i 's/#user_allow_other/user_allow_other/g' /etc/fuse.conf || (sudo sh -c 'echo "user_allow_other" > /etc/fuse.conf');
```
Comment (romilbhardwaj) [inline sky/templates/aws-ray.yml.j2:189]: Can we let the comment stay?

File: sky/templates/aws-ray.yml.j2 [hunk 2]
```diff
@@ -185,7 +186,12 @@ setup_commands:
...
+    {%- if remote_identity != 'LOCAL_CREDENTIALS' %}
+    rm ~/.aws/credentials || true;
```
Comment (romilbhardwaj) [inline sky/templates/aws-ray.yml.j2:191]: Say a user manually logged in a configured AWS with `aws configure`. Would a new `sky launch` then delete their custom added credentials?

File: sky/templates/aws-ray.yml.j2 (hunk 1 above)
Comment (Michaelvll) [reply sky/templates/aws-ray.yml.j2:189]: The comment is already included in the comments at the top : )

The inline comment does not work with the setup commands.

File: sky/templates/aws-ray.yml.j2 (hunk 2 above)
Comment (Michaelvll) [reply sky/templates/aws-ray.yml.j2:191]: Yes, we will delete their custom-added credentials. A user setting up `aws configure` themselves on the remote cluster can be an undefined behavior, as that could break our autostop as well.

A better solution is to make the credentials declarative, i.e. note the credential/active service account for SkyPilot in the cluster yaml and use the credential everywhere in our backend. This is a temp solution to unblock our users. Wdyt?

Comment (romilbhardwaj) [reply sky/templates/aws-ray.yml.j2:191]: sgtm!

**Pull request merged**